mesonbuild/dependencies/dev.py

```diff
@@ -549,6 +552,18 @@ def __machine_info_to_platform_include_dir(m: 'MachineInfo') -> T.Optional[str]:
         return None
 
 
+class JDKSystemDependency(JNISystemDependency):
+    def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
+        super().__init__(environment, kwargs)
```

Comment:
This probably results in the jdk dependency having a `dep.name()` of `jni` even when you look it up with `jdk`.

This is... probably fine? It is a deprecated alias and all that.

Reply:
I figured it would probably be a little painful in that regard, but since it is deprecated, I don't think anyone will Notice. I could do a refactor however and move the actual lookups to a common function instead.

Reply:
I agree that it may be entirely too painful to bother for a deprecated thing.